src/compiler/fbc.bas

```diff
@@ -3114,10 +3114,18 @@ private function hCompileStage2Module( byval module as FBCIOFILE ptr ) as intege
 		end select
 
 		if( fbGetOption( FB_COMPOPT_TARGET ) <> FB_COMPTARGET_JS ) then
-			if( fbc.cputype_is_native ) then
-				ln += "-march=native "
+			if( fbGetCpuFamily( ) = FB_CPUFAMILY_PPC or FB_CPUFAMILY_PPC64 ) then
+				if( fbc.cputype_is_native ) then
+					ln += "-mcpu=native "
+				else
+					ln += "-mcpu=" + *fbGetGccArch( ) + " "
+				end if
```

Comment:
Firstly, `FB_CPUFAMILY_PPC or FB_CPUFAMILY_PPC64` is a mistake.

I was surprised by this switch to -mcpu, so I looked into it. I am shocked and appalled to find that the meaning and availability of the `-march`/`-mtune`/`-mcpu` arguments to gcc are target-specific. The PPC target doesn't even recognise `-march`, and they are especially complicated on ARM. Clang [tries to copy](https://groups.google.com/g/llvm-dev/c/oNN5Ob2Dc6I/m/j3i1iiadCwAJ) GCC's insane argument scheme, but other LLVM utilities do not.

It would be very helpful to have a note about that in a comment.

Reply:
Also, `-mcpu=... -mtune=...` would be more closely equivalent to the `-march=...` used for other targets. [E.g. this](https://github.com/mfem/mfem/issues/216).

Reply:
@rversteegen Good to hear from you.  These are your headers, that you pointed me to.  I apologize for not discussing this with you.  Some cut and pastes were made in response to real world testing that I mentioned in #290.

Reply:
Ah, so you did. Thanks for adding the comment.

Also, the mistake I was referring to was
``if( fbGetCpuFamily( ) = FB_CPUFAMILY_PPC or FB_CPUFAMILY_PPC64 ) then``
(which github unfortunately has cut out of the preview on the Conversation tab) should be
``if( (fbGetCpuFamily( ) = FB_CPUFAMILY_PPC) orelse (fbGetCpuFamily( ) = FB_CPUFAMILY_PPC64) ) then``

Reply:
I’ll make the changes and resubmit the PR.

Reply:
Comment added to code explaining reason for change